### PR TITLE
152 refactoring fix

### PR DESCRIPTION
**Pull request by katethemate**

### CODE REVIEW

File: SLAM/core.py [hunk 1]
```diff
@@ -43,3 +43,10 @@ def __eq__(self, other):
         else:
             return False
 
+    def transfer_coordinates(self,landmark):
```
Comment (davidlkang) [inline SLAM/core.py]: Could you decide on some uniform name for a second landmark?
In the `__eq__` function you use `other` and here `landmark`.

File: SLAM/globalizer.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+#translates local coordinates to global coordinates
+def globalize_points(robot_position,local_coordinates):
+    global_coordinates = []
+    for local_coordinate in local_coordinates:
+        local_coordinate = local_coordinate+robot_position
```
Comment (davidlkang) [inline SLAM/globalizer.py]: Could you not overwrite this variable, but create a new one.

File: SLAM/globalizer.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        global_coordinates.append(local_coordinate)
+    return global_coordinates
+
+def globalize_landmarks(robot_position,local_landmarks):
+    global_landmarks = []
+    for local_landmark in local_landmarks:
+        local_landmark.A = local_landmark.A+robot_position
+        local_landmark.B = local_landmark.B+robot_position
+        local_landmark.C = local_landmark.C+robot_position
```
Comment (davidlkang) [inline SLAM/globalizer.py:15]: The same here as above.

File: SLAM/main.py [hunk 4]
```diff
@@ -16,25 +16,28 @@ def create_points(point_batch):
...
+        stored_landmarks.append(new_landmarks)
+        send_points(point_cloud)
         send_position(robot_position)
 
 
-    return stored_landmarks, new_global_landmarks, global_points
-
-
-
+    return stored_landmarks, new_landmarks, point_cloud
```
Comment (davidlkang) [inline SLAM/main.py]: This file is redundant now.

File: main.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import SLAM.core
+from SLAM.landmark_extractor import extract_spike_landmarks
+from SLAM.landmark_matcher import match_landmarks
+from SLAM.position_evaluator import evaluate_position
+from SLAM.receiver import Receiver
+from multiprocessing import Queue
+from SLAM.globalizer import *
+from SLAM.sender import *
```
Comment (davidlkang) [inline main.py]: Can you import the functions you want to use directly, instead of importing everything.

File: main.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        point_cloud = create_points(point_batch) #local points
+        found_landmarks = extract_spike_landmarks(point_cloud)   #local landmarks
+        matched_landmarks,new_landmarks = match_landmarks(found_landmarks, stored_landmarks)
+
+        if first_iteration:
+            robot_position= Point(None,None,x=0,y=0)
+        else:
+            robot_position = evaluate_position(matched_landmarks)
+            new_landmarks = globalize_landmarks(robot_position, new_landmarks)
+            point_cloud = globalize_points(robot_position,point_cloud)
```
Comment (davidlkang) [inline main.py]: Is this ever called? As far as I see it, `first_iteration` will never be falsy.

**Review by davidlkang — CHANGES_REQUESTED** [review]

File: main.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+from SLAM.receiver import Receiver
+from multiprocessing import Queue
+from SLAM.globalizer import globalize_points, globalize_landmarks
+from SLAM.sender import send_points, send_position
+
+# creates point objects
+def create_points(point_batch):
+    points = []
+    for p in point_batch:
+        points.append(SLAM.core.Point(p[0], p[1]))
```
Comment (elli-wolf) [inline main.py]:
```suggestion
        points.append(Point(p[0], p[1]))
```
You never imported SLAM itself, therefore this line will fail.

**Review by davidlkang — APPROVED** [review]

File: SLAM/core.py [hunk 8]
```diff
@@ -29,17 +39,43 @@ def __str__(self):
...
 
+    def transfer_coordinates(self,other_landmark):
+        self.A.x = other_landmark.A.x
+        self.A.y = other_landmark.A.y
+        self.B.x = other_landmark.B.x
+        self.B.y = other_landmark.B.y
+        self.C.x = other_landmark.C.x
+        self.C.y = other_landmark.C.y
+
+    def get_avg_robot_poistion(self):
```
Comment (davidlkang) [inline SLAM/core.py]:
```suggestion
    def get_avg_robot_position(self):
```

File: SLAM/core.py [hunk 9]
```diff
@@ -29,17 +39,43 @@ def __str__(self):
...
+        alpha_A = calculate_alpha(self.A.rotation)
+        ZAx = math.sin(alpha_A)*self.A.distance
+        ZAy = math.cos(alpha_A)*self.A.distance
```
Comment (davidlkang) [inline SLAM/core.py:67]: Is this an object? Because the first letter is capital.

File: SLAM/position_evaluator.py [hunk 10]
```diff
@@ -2,34 +2,11 @@
...
-    elif 90<stepper_angle<=180:
-        return math.radians(180-stepper_angle)
-    elif 180<stepper_angle<=270:
-        return math.radians(270-stepper_angle)
-    else:
-        return math.radians(360-stepper_angle)
+def evaluate_avg_position(landmarks):
+    point_sum = Point(None, None, 0, 0)
+    for l in landmarks:
+        point_sum = point_sum + l.get_avg_robot_poistion()
```
Comment (davidlkang) [inline SLAM/position_evaluator.py]:
```suggestion
        point_sum = point_sum + l.get_avg_robot_position()
```

**Pull request merged**